src/ai/ai_scanner.cpp

```diff
@@ -65,31 +66,21 @@ AIInfo *AIScannerInfo::SelectRandomAI() const
 		return this->info_dummy;
 	}
 
-	uint num_random_ais = 0;
-	for (const auto &item : info_single_list) {
-		AIInfo *i = static_cast<AIInfo *>(item.second);
-		if (i->UseAsRandomAI()) num_random_ais++;
-	}
+	/* Filter for AIs suitable as Random AI. */
+	auto random_ais = info_single_list | std::views::filter([](const auto &item) { return static_cast<AIInfo *>(item.second)->UseAsRandomAI(); });
 
+	uint num_random_ais = std::ranges::distance(random_ais);
 	if (num_random_ais == 0) {
```

Comment:
not worth keeping this as a variable
```suggestion
	if (std::ranges::empty(random_ais)) {
```

Reply:
Nope, missed the GetRandomizer call. Disregard.

Reply:
It's used on line 79 to pick a random number.